### PR TITLE
fix json patch issue

### PR DESCRIPTION
**Pull request by zhaohuabing**
fix https://github.com/envoyproxy/gateway/pull/1628#discussion_r1255191062

It's a bit tricky because Marshal method actually converts the json in the operation value to a string literal by adding escape characters. The workaround is manually creating the json patch.

### CODE REVIEW

**Comment by zhaohuabing** [comment]
why coverage failed?

File: internal/xds/translator/testdata/in/xds-ir/jsonpatch.yaml [hunk 1]
```diff
@@ -21,15 +21,15 @@ jsonPatches:
...
+    value: |
+      - actions:
+        - remote_address: {}
 - type: "type.googleapis.com/envoy.config.cluster.v3.Cluster"
   name: rate-limit-cluster
   operation:
     op: add
     path: ""
-    value:
+    value: |
```
Comment (zirain) [inline internal/xds/translator/testdata/in/xds-ir/jsonpatch.yaml:32]: is the '|' required? if so, I don't think this is better than before.

Comment (zhaohuabing) [reply internal/xds/translator/testdata/in/xds-ir/jsonpatch.yaml:32]: The  "|" means that the value is a multi-line string, that's it.

Comment (zirain) [reply internal/xds/translator/testdata/in/xds-ir/jsonpatch.yaml:32]: It's ok for a configmap.
I think here we want some like `pb.Struct` or `map[string]interface{}`, so I prefer `apiext.JSON` than `string`.

Comment (zhaohuabing) [reply internal/xds/translator/testdata/in/xds-ir/jsonpatch.yaml:32]: The value of the Patch is a string. Besides, it's very hard to troubleshooting if we define the API as a byte array.

Comment (zhaohuabing) [reply internal/xds/translator/testdata/in/xds-ir/jsonpatch.yaml:32]: I does look a bit weird. It's better to just treat the value as a yaml/json structure.

**Pull request closed without merging**